event.js

```diff
@@ -1,12 +1,20 @@
-
 module.exports = function awaitEvent({ eventEmitter, resolveOn = [], rejectOn = [], timeout = Infinity, forceArray = false })
 {
+    if (!eventEmitter)
+        throw new Error("No event emitter provided");
+
     const events = [].concat(resolveOn);
-    const errorEvents = [].concat(rejectOn)
+    const errorEvents = [].concat(rejectOn);
+
+    if (events.length === 0 && errorEvents.length === 0)
+        throw new Error("No events or errors provided. This makes the promise unsettleable");
```

Comment:
👍 (Maybe "error events", error message [how meta] could just be slightly clearer.)